app/forms/AtedForms.scala

```diff
@@ -41,48 +43,55 @@ object AtedForms {
   val SIXTY = 60
   val numRegex = """[0-9]{8}"""
   val addressLineLength = 35
-  val postcodeLength = 9
+  val postcodeLength = 10
   val countryLength = 2
-  val emailLength = 241
+  val emailLength = 132
   val lengthZero = 0
   val nameLength = 35
   val phoneLength = 24
   val faxLength = 24
   val businessNameLength = 105
-  val telephoneRegex = """^[A-Z0-9)\/(\-*#]+$""".r
+  val businessNameRegex = "^[a-zA-Z0-9 '&\\\\/]{1,105}$"
+  val telephoneRegex = "^[A-Z0-9 )/(\\-*#]+$".r
   val emailRegex =
     """^(?!\.)("([^"\r\\]|\\["\r\\])*"|([-a-zA-Z0-9!#$%&'*+/=?^_`{|}~]|(?<!\.)\.)*)
       |(?<!\.)@[a-zA-Z0-9][\w\.-]*[a-zA-Z0-9]\.[a-zA-Z][a-zA-Z\.]*[a-zA-Z]$""".r
-  val postCodeRegex = """^[A-Z]{1,2}[0-9R][0-9A-Z]? [0-9][ABD-HJLNP-UW-Z]{2}$""".r
-  val addressLineRegex = """^([A-Za-z0-9\s\&\-\,\@\#\.\'])*$""".r
+  val postCodeRegex = "^[A-Z]{1,2}[0-9][0-9A-Z]?\\s?[0-9][A-Z]{2}|BFPO\\s?[0-9]{1,10}$".r
+  val addressLineRegex = "^[A-Za-z0-9 \\-,.&']{1,35}$".r
+  val addressRegex = "^[A-Za-z0-9 \\-,.&']{1,35}$"
+  val nameRegex = "^[a-zA-Z &`\\-\'^]{1,35}$"
```

Comment:
Please follow best practices to name constants. Would it be worth moving these to constant file?

Reply:
Fixed it